### ISSUE TITLE
'strict' and 'existing' flags not working from config file

### ISSUE THREAD

**Issue by martingleria** [lukaszraczylo/semver-generator#13]
Hi!

Let me spend one line making an explicit **thank you** for sharing your knowledge and time in this project. 

Now, to the issue:

I was testing your tool to incorporate it into our workflows. I found that the recently added `existing: true` flag, and also the _strict_ flag don't work from the config file. 

This is my config file:

```
version: 1
force:
  existing: true
  strict: true
wording:
  patch:
    - update
    - fix
  minor:
    - feature
    - improve
  major:
    - breaking
    - major
  release:
    - release-candidate
    - add-rc
```

I was originally testing directly from a Github Action:

```
name: Testing semver

on:
  push:
    branches:
      - main

jobs:
  prepare:
    name: Preparing build context
    runs-on: ubuntu-latest
    outputs:
      RELEASE_VERSION: ${{ steps.semver.outputs.semantic_version }}
    steps:
      - name: Checkout repo
        uses: actions/checkout@v2
        with:
          fetch-depth: '0'
      - name: Semver run
        id: semver
        uses: lukaszraczylo/semver-generator@v1
        with:
          config_file: semver.yml
          repository_local: true
      - name: Semver check
        run: |
          echo "Semantic version detected: ${{ steps.semver.outputs.semantic_version }}"
```

The result was that existing tags were not taken into account, and strict mode didn't work either. 

I download the project to test it locally and I realized that from my local computer I'm able to reproduce the same behavior running the binary in this way (semver.yml is the same one I included before):

```
./semver-gen generate -l  -c semver.yml
SEMVER 0.0.13
```

But if I run explicitly including the flags the result is the desired one:

````
./semver-gen generate -l -s -e -c semver.yml
SEMVER 0.1.0
````

Am I doing something wrong? Otherwise, I hope this requires minimal effort on your side since it's all I need for my workflows right now. 

Thanks again.

**Comment by lukaszraczylo** [comment]
Hi @martingleria - let me have a look :)

**Comment by lukaszraczylo** [comment]
Could you try updating the semver-gen binary to the latest version and try it again? :)

**Comment by martingleria** [comment]
Something definitively changed, but I'm not sure if the current one is the right result. By combining _strict_ and _existent_ flags, I expect that the sem-ver calculated was 0.1.0, like the last output I shared.

However, now I'm getting 0.1.3. Check it out:

````
./semver-gen generate -l -d -c semver.yml
DEBUG: 
---COMMITS BEFORE CUT---
 []

DEBUG: 
---COMMITS AFTER CUT---
 [{290e15fc91cd804861d81038b1425ec09134d9e3 Martin Gleria <martin.gleria@agileengine.com> first commit
 2022-02-07 19:27:05 +0100 +0100} {271f04e8ba509d54fd598fbf70738e432112606e Martin Gleria <martin.gleria@agileengine.com> Change 1
 2022-02-07 19:28:19 +0100 +0100} {9bfc247b576b8ca3010ac098ddc7a4bfc68a94cf Martin Gleria <martin.gleria@agileengine.com> commit without special words
 2022-02-07 19:53:41 +0100 +0100} {406bd36dcee952903bada41141325934bd3f4f63 Martin Gleria <martin.gleria@agileengine.com> simple commit
 2022-02-07 19:55:11 +0100 +0100} {bf46499cd0d64097e2ca97e0b55298337e823c6a Martin Gleria <martin.gleria@agileengine.com> workflow on the right path
 2022-02-07 20:04:10 +0100 +0100} {a96d478166d0910ebac5bca840651624f09a38a7 Martin Gleria <martin.gleria@agileengine.com> Removing force block
 2022-02-08 17:48:10 +0100 +0100} {ca985a1ab74454dbcf8ab70c6005b28e89010948 Martin Gleria <martin.gleria@agileengine.com> repository_local: true
 2022-02-08 18:07:22 +0100 +0100} {99e8153c7b511ca51e61b37047673ced9b9a8dd5 Martin Gleria <martin.gleria@agileengine.com> special word: fix
 2022-02-08 18:13:53 +0100 +0100} {69d7b3743696f01893dd1fd4af7ca040935bf20b Martin Gleria <martin.gleria@agileengine.com> Testing force again
 2022-02-08 18:17:21 +0100 +0100} {d98553b5419f3fe575c9e94af4de4be4ca320d5c Martin Gleria <martin.gleria@agileengine.com> existing: true
 2022-02-08 19:49:56 +0100 +0100} {a9ba7176c30c53635f226e539caf890a72ae55f5 Martin Gleria <martin.gleria@agileengine.com> Checking existent version tag
 2022-02-08 19:52:55 +0100 +0100} {a328d5c32b70139c34c9321746cb5dd40d5bfde2 Martin Gleria <martin.gleria@agileengine.com> testing strict
 2022-02-08 20:45:49 +0100 +0100}]

DEBUG: Listing existing tags

DEBUG: Found tag: v0.0.1 271f04e8ba509d54fd598fbf70738e432112606e

DEBUG: Found tag: v0.1.0 d98553b5419f3fe575c9e94af4de4be4ca320d5c

DEBUG: Incrementing patch (DEFAULT) on  first commit | Semver: 0.0.1

DEBUG: Found existing tag: v0.0.1 related to Change 1


DEBUG: Parsing existing semver: v0.0.1

DEBUG: Incrementing patch (DEFAULT) on  Change 1 | Semver: 0.0.2

DEBUG: Incrementing patch (DEFAULT) on  commit without special words | Semver: 0.0.3

DEBUG: Incrementing patch (DEFAULT) on  simple commit | Semver: 0.0.4

DEBUG: Incrementing patch (DEFAULT) on  workflow on the right path | Semver: 0.0.5

DEBUG: Incrementing patch (DEFAULT) on  Removing force block | Semver: 0.0.6

DEBUG: Incrementing patch (DEFAULT) on  repository_local: true | Semver: 0.0.7

DEBUG: Incrementing patch (DEFAULT) on  special word: fix | Semver: 0.0.8

DEBUG: Found match for  fix | fix

DEBUG: Incrementing patch (WORDING) on  special word: fix | Semver: 0.0.9

DEBUG: Incrementing patch (DEFAULT) on  Testing force again | Semver: 0.0.10

DEBUG: Found existing tag: v0.1.0 related to existing: true


DEBUG: Parsing existing semver: v0.1.0

DEBUG: Incrementing patch (DEFAULT) on  existing: true | Semver: 0.1.1

DEBUG: Incrementing patch (DEFAULT) on  Checking existent version tag | Semver: 0.1.2

DEBUG: Incrementing patch (DEFAULT) on  testing strict | Semver: 0.1.3

SEMVER 0.1.3
````
But If I use all the flags from the command line, I get the 0.1.0 desired:

````
./semver-gen generate -l -e -s -d -c semver.yml
DEBUG: 
---COMMITS BEFORE CUT---
 []

DEBUG: 
---COMMITS AFTER CUT---
 [{290e15fc91cd804861d81038b1425ec09134d9e3 Martin Gleria <martin.gleria@agileengine.com> first commit
 2022-02-07 19:27:05 +0100 +0100} {271f04e8ba509d54fd598fbf70738e432112606e Martin Gleria <martin.gleria@agileengine.com> Change 1
 2022-02-07 19:28:19 +0100 +0100} {9bfc247b576b8ca3010ac098ddc7a4bfc68a94cf Martin Gleria <martin.gleria@agileengine.com> commit without special words
 2022-02-07 19:53:41 +0100 +0100} {406bd36dcee952903bada41141325934bd3f4f63 Martin Gleria <martin.gleria@agileengine.com> simple commit
 2022-02-07 19:55:11 +0100 +0100} {bf46499cd0d64097e2ca97e0b55298337e823c6a Martin Gleria <martin.gleria@agileengine.com> workflow on the right path
 2022-02-07 20:04:10 +0100 +0100} {a96d478166d0910ebac5bca840651624f09a38a7 Martin Gleria <martin.gleria@agileengine.com> Removing force block
 2022-02-08 17:48:10 +0100 +0100} {ca985a1ab74454dbcf8ab70c6005b28e89010948 Martin Gleria <martin.gleria@agileengine.com> repository_local: true
 2022-02-08 18:07:22 +0100 +0100} {99e8153c7b511ca51e61b37047673ced9b9a8dd5 Martin Gleria <martin.gleria@agileengine.com> special word: fix
 2022-02-08 18:13:53 +0100 +0100} {69d7b3743696f01893dd1fd4af7ca040935bf20b Martin Gleria <martin.gleria@agileengine.com> Testing force again
 2022-02-08 18:17:21 +0100 +0100} {d98553b5419f3fe575c9e94af4de4be4ca320d5c Martin Gleria <martin.gleria@agileengine.com> existing: true
 2022-02-08 19:49:56 +0100 +0100} {a9ba7176c30c53635f226e539caf890a72ae55f5 Martin Gleria <martin.gleria@agileengine.com> Checking existent version tag
 2022-02-08 19:52:55 +0100 +0100} {a328d5c32b70139c34c9321746cb5dd40d5bfde2 Martin Gleria <martin.gleria@agileengine.com> testing strict
 2022-02-08 20:45:49 +0100 +0100}]

DEBUG: Listing existing tags

DEBUG: Found tag: v0.0.1 271f04e8ba509d54fd598fbf70738e432112606e

DEBUG: Found tag: v0.1.0 d98553b5419f3fe575c9e94af4de4be4ca320d5c

DEBUG: Found existing tag: v0.0.1 related to Change 1


DEBUG: Parsing existing semver: v0.0.1

DEBUG: Found match for  fix | fix

DEBUG: Incrementing patch (WORDING) on  special word: fix | Semver: 0.0.2

DEBUG: Found existing tag: v0.1.0 related to existing: true


DEBUG: Parsing existing semver: v0.1.0

SEMVER 0.1.0
````

**Comment by lukaszraczylo** [comment]
I'm trying to reproduce your repository and issue. Could you:

- make sure you have `v1.4.6` of the semver-gen binary
- the `strict: true` is set in the config under `force.strict: true`
- if both above are fine - could you please paste the config so I can re-create your repository and run few checks on my side :)

Thank you so much for your patience and I really appreciate your help!

**Comment by martingleria** [comment]
It's working now! Thank you!

Apparently, I pulled the code between your last two commits from yesterday, so I was missing one of them.

I tested directly on Github Actions and the result was the expected:
<img width="453" alt="image" src="https://user-images.githubusercontent.com/92102535/153621506-92a52acf-1c69-44c3-9b16-25e19221a72f.png">


Immediately after, I pushed a new commit with one of the special words ('improve') I have on the config file, for minor changes. As a result, I was expecting to see 0.2.0, but I got instead 0.2.1. 

Am I interpreting correctly? What do you think of this result:

````

./semver-gen generate -l -d -c semver.yml
DEBUG: 
---COMMITS BEFORE CUT---
 []

DEBUG: 
---COMMITS AFTER CUT---
 [{290e15fc91cd804861d81038b1425ec09134d9e3 Martin Gleria <martin.gleria@agileengine.com> first commit
 2022-02-07 19:27:05 +0100 +0100} {271f04e8ba509d54fd598fbf70738e432112606e Martin Gleria <martin.gleria@agileengine.com> Change 1
 2022-02-07 19:28:19 +0100 +0100} {9bfc247b576b8ca3010ac098ddc7a4bfc68a94cf Martin Gleria <martin.gleria@agileengine.com> commit without special words
 2022-02-07 19:53:41 +0100 +0100} {406bd36dcee952903bada41141325934bd3f4f63 Martin Gleria <martin.gleria@agileengine.com> simple commit
 2022-02-07 19:55:11 +0100 +0100} {bf46499cd0d64097e2ca97e0b55298337e823c6a Martin Gleria <martin.gleria@agileengine.com> workflow on the right path
 2022-02-07 20:04:10 +0100 +0100} {a96d478166d0910ebac5bca840651624f09a38a7 Martin Gleria <martin.gleria@agileengine.com> Removing force block
 2022-02-08 17:48:10 +0100 +0100} {ca985a1ab74454dbcf8ab70c6005b28e89010948 Martin Gleria <martin.gleria@agileengine.com> repository_local: true
 2022-02-08 18:07:22 +0100 +0100} {99e8153c7b511ca51e61b37047673ced9b9a8dd5 Martin Gleria <martin.gleria@agileengine.com> special word: fix
 2022-02-08 18:13:53 +0100 +0100} {69d7b3743696f01893dd1fd4af7ca040935bf20b Martin Gleria <martin.gleria@agileengine.com> Testing force again
 2022-02-08 18:17:21 +0100 +0100} {d98553b5419f3fe575c9e94af4de4be4ca320d5c Martin Gleria <martin.gleria@agileengine.com> existing: true
 2022-02-08 19:49:56 +0100 +0100} {a9ba7176c30c53635f226e539caf890a72ae55f5 Martin Gleria <martin.gleria@agileengine.com> Checking existent version tag
 2022-02-08 19:52:55 +0100 +0100} {a328d5c32b70139c34c9321746cb5dd40d5bfde2 Martin Gleria <martin.gleria@agileengine.com> testing strict
 2022-02-08 20:45:49 +0100 +0100} {f46d54bfb473abaa59158cd3eba7c35445d82d6d Martin Gleria <martin.gleria@agileengine.com> Testing new action version
 2022-02-11 16:23:16 +0100 +0100} {66d0a4218c1ce06cf123f479f807d53f3df4c2ce Martin Gleria <martin.gleria@agileengine.com> testing improve key word
 2022-02-11 16:31:29 +0100 +0100}]

DEBUG: Listing existing tags

DEBUG: Found tag: v0.0.1 271f04e8ba509d54fd598fbf70738e432112606e

DEBUG: Found tag: v0.1.0 d98553b5419f3fe575c9e94af4de4be4ca320d5c

DEBUG: Found existing tag: v0.0.1 related to Change 1


DEBUG: Parsing existing semver: v0.0.1

DEBUG: Found match for  fix | fix

DEBUG: Incrementing patch (WORDING) on  special word: fix | Semver: 0.0.2

DEBUG: Found existing tag: v0.1.0 related to existing: true


DEBUG: Parsing existing semver: v0.1.0

DEBUG: Found match for  improve | improve

DEBUG: Incrementing minor (WORDING) on  testing improve key word | Semver: 0.2.1

SEMVER 0.2.1

**Comment by lukaszraczylo** [comment]
Wonderful, happy to hear that - I will double-check this evening. It could be related to the way semver is calculated _after_ the existing tag is detected :)
